packages/plugin-electron-session/session.js

```diff
@@ -1,12 +1,39 @@
 const sessionDelegate = require('@bugsnag/plugin-browser-session')
+const Session = require('@bugsnag/core/session')
 
 const SESSION_TIMEOUT_MS = 60 * 1000
 
-module.exports = (app, BrowserWindow) => ({
+module.exports = (app, BrowserWindow, NativeClient) => ({
   load (client) {
+    // Ensure native session kept in sync with session changes
+    Session.prototype._track = new Proxy(Session.prototype._track, {
+      apply (target, _this, args) {
+        const result = target.apply(_this, args)
+        NativeClient.setSession(serializeForNativeEvent(_this))
+        return result
+      }
+    })
```

Comment:
I guess really this might be overcomplicating things 😅 We could use the same approach as the session delegate:

```suggestion
    // Ensure native session kept in sync with session changes
    const oldTrack = Session.prototype._track
    Session.prototype._track = function (...args) {
      const result = oldTrack.apply(this, args)
      NativeClient.setSession(serializeForNativeEvent(this))
      return result
    }
```
I'm happy either way though, it doesn't really make a difference 🤷